scripts/create-initial-changelog.sh

```diff
@@ -0,0 +1,38 @@
+#-*- coding: utf-8-unix -*-
+#!/usr/bin/env bash
+
+# Initialize the changelog file
+filename='CHANGELOG.md'
+
+echo "# Changelog" > $filename
+echo "This is the changelog for *casual java* and all changes are listed in this document." >> $filename
+echo "" >> $filename
+
+# Loop through all tags, sorted by version number (newest first)
+for tag in $(git tag --sort=-version:refname); do
+    # Get the commit hash for the tag
+    commit_hash=$(git rev-list -n 1 "$tag")
+    
+    # Get the commit message (summary)
+    commit_message=$(git log -1 --pretty=format:"%s" "$commit_hash")
+    
+    # Get the full commit message (body)
+    full_commit_message=$(git log -1 --pretty=format:"%b" "$commit_hash")
+    
+    # Get the commit date in YYYY-MM-DD format
+    commit_date=$(git log -1 --pretty=format:"%cd" --date=short "$commit_hash")
+    
+    # Write the tag and commit details to the changelog
+    echo "## [$tag] - $commit_date" >> $filename
+    echo "- $commit_message" >> $filename
+
+    # Remove leading '*' and whitespace
+    modified_full_commit_message=$(echo "$full_commit_message" | sed 's/^[*[:space:]]*//')
```

Comment:
Why remove leading '*'?
For example in this commit:
https://github.com/casualcore/casual-java/commit/62ce53a7c299c113dd5efd1bce168a7ba229893c
I have a list of changes - this would make the first items not part of the list.

Reply:
https://github.com/casualcore/casual-java/blob/feature/not-dev/CHANGELOG.md

Just tested it, it works as it should.
All * items from the PR remains.

Reply:
https://github.com/casualcore/casual-java/blob/feature/not-dev/CHANGELOG.md#3247---2024-11-21

If you look here, you can see it seems to have added an extra one bullet for the message summary and removed all bar the first one for the description. Rather than maintain them as a list.